graph/api/src/main/java/org/jboss/windup/graph/model/HasApplications.java

```diff
@@ -1,30 +1,27 @@
-package org.jboss.windup.rules.apps.javaee.model;
+package org.jboss.windup.graph.model;
 
 import com.tinkerpop.frames.modules.javahandler.JavaHandler;
-import org.jboss.windup.graph.model.BelongsToProject;
-import org.jboss.windup.graph.model.ProjectModel;
 
 import com.tinkerpop.blueprints.Vertex;
 import com.tinkerpop.frames.modules.javahandler.JavaHandlerContext;
 
-import java.util.HashSet;
-import java.util.Set;
-
 /**
  * @author <a href="mailto:dklingenberg@gmail.com">David Klingenberg</a>
  */
 public interface HasApplications extends BelongsToProject
 {
+    /**
+     * Gets all root project models for current model (This will be mostly 1, but there are few exceptions which have multiple project models, so it
+     * returns Iterable to keep interface consistent)
+     *
+     * @return root project models
+     */
     Iterable<ProjectModel> getApplications();
```

Comment:
If I understand it correctly and it really returns the parent project models, then the name of interface probably shouldn't be `HasApplications` but `IsPartOfApplications` or something like that.

`getApplications()` in combination with `HasApplications` interface would imply it returns child applications, not parent ones.

Reply:
It isn't always accurate that it is a part of an application, though. For example, if we find a reference to a datasource in an application, that means that datasource was used by that application. It doesn't necessarily follow that the datasource is defined there.

HasApplications just implies that a link should be present, not that an ownership relationship necessarily exists.